plugins/modules/dns_auth_nsg_info.py

```diff
@@ -0,0 +1,386 @@
+            type: list
+            returned: Always
+        tags:
+            description:
+                - "Tagging specifics."
+            type: dict
+            returned: Always
+"""  # noqa: E501
+
+from ansible_collections.infoblox.bloxone.plugins.module_utils.modules import BloxoneAnsibleModule
+
+try:
+    from bloxone_client import ApiException, NotFoundException
+    from dns_config import AuthNsgApi
+except ImportError:
+    pass  # Handled by BloxoneAnsibleModule
+
+
+class AuthNsgInfoModule(BloxoneAnsibleModule):
+    def __init__(self, *args, **kwargs):
+        super(AuthNsgInfoModule, self).__init__(*args, **kwargs)
+        self._existing = None
+        self._limit = 1000
+
+    def find_by_id(self):
+        try:
+            resp = AuthNsgApi(self.client).read(self.params["id"], inherit="full")
+            return [resp.result]
+        except NotFoundException as e:
+            return None
+
+    def find(self):
+        if self.params["id"] is not None:
+            return self.find_by_id()
+
+        filter_str = None
+        if self.params["filters"] is not None:
+            filter_str = " and ".join([f"{k}=='{v}'" for k, v in self.params["filters"].items()])
+        elif self.params["filter_query"] is not None:
+            filter_str = self.params["filter_query"]
+
+        tag_filter_str = None
+        if self.params["tag_filters"] is not None:
+            tag_filter_str = " and ".join([f"{k}=='{v}'" for k, v in self.params["tag_filters"].items()])
+        elif self.params["tag_filter_query"] is not None:
+            tag_filter_str = self.params["tag_filter_query"]
+
+        all_results = []
+        offset = 0
+
+        while True:
+            try:
+                resp = AuthNsgApi(self.client).list(
+                    offset=offset, limit=self._limit, filter=filter_str, tfilter=tag_filter_str
+                )
+                all_results.extend(resp.results)
+
+                if len(resp.results) < self._limit:
+                    break
+                offset += self._limit
+
+            except ApiException as e:
+                self.fail_json(msg=f"Failed to execute command: {e.status} {e.reason} {e.body}")
+
+        return all_results
+
+    def run_command(self):
+        result = dict(objects=[])
+
+        if self.check_mode:
+            self.exit_json(**result)
+
+        find_results = self.find()
+
+        all_results = []
+        for r in find_results:
+            all_results.append(r.model_dump(by_alias=True, exclude_none=True))
+
+        result["objects"] = all_results
+        self.exit_json(**result)
+
+
+def main():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict(
+        id=dict(type="str", required=False),
+        filters=dict(type="dict", required=False),
+        filter_query=dict(type="str", required=False),
+        inherit=dict(type="str", required=False, choices=["full", "partial", "none"], default="full"),
```

Comment:
Check for inheritance support !

Reply:
not supported, removed and tested the playbook